### PR TITLE
Add an AbortSignal to the request options

### PR DESCRIPTION
**Pull request by cbiesinger**
***
<a href="https://pr-preview.s3.amazonaws.com/cbiesinger/WebID/pull/162.html" title="Last updated on Dec 10, 2021, 5:36 PM UTC (f46f5e5)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/FedCM/162/f25f0d4...cbiesinger:f46f5e5.html" title="Last updated on Dec 10, 2021, 5:36 PM UTC (f46f5e5)">Diff</a>

### CODE REVIEW

**Comment by cbiesinger** [comment]
@dj2 thoughts?

File: spec/index.bs [hunk 1]
```diff
@@ -887,11 +887,12 @@ partial interface FederatedCredential {
     ::  When used in conjunction with {{CredentialRequestOptions/mediation}}, it indicates whether the credential was provided via a [[#use-cases-auto-sign-in]] or an [[#use-cases-explicit-sign-in]].
 </dl>
 
-Second, it extends the {{FederatedCredentialRequestOptions}} by adding a list of {{FederatedIdentityProvider}}s:
+Second, it extends the {{FederatedCredentialRequestOptions}} by adding a list of {{FederatedIdentityProvider}}s and a way to abort the get request:
```
Comment (dj2) [inline spec/index.bs:890]: Maybe just "and a way to abort the request" ?

File: spec/index.bs [hunk 2]
```diff
@@ -1044,6 +1045,9 @@ To <dfn>mint</dfn> a new id token, run the following algorithm:
     * {{FederatedCredential/idToken}} with |token|
     * {{FederatedCredential/provider}} with |provider|["{{FederatedIdentityProvider/url}}"]
 
+If a signal is provided in the request options, and the signal gets aborted before the get request
+finishes, the promise must be rejected with a DOMException containing an AbortError.
```
Comment (dj2) [inline spec/index.bs:1049]: Which direction does this abort signal go? Is it to signal to the RP that the get was aborted? Or is it to allow the get to be aborted by the RP? (Or, maybe I'm mis-reading your statement?)

The `AbortSignal` on CredentialRequestOption says:
```
This property lets the developer abort an ongoing get() operation. An aborted operation may complete normally (generally if the abort was received after the operation finished) or reject with an "AbortError" DOMException."
```

Which seems like it's the RP aborting the get?

**Comment by cbiesinger** [comment]
Turns out there's already a signal in https://w3c.github.io/webappsec-credential-management/#credentialrequestoptions-dictionary, so I'm withdrawing this pull request.

**Comment by samuelgoto** [comment]
Did I not send this to you earlier? If not, terribly sorry, it was on my queue and I don't think I clicked the right buttons :)

**Comment by cbiesinger** [comment]
@samuelgoto I don't think it got sent until now but github's review UI constantly confuses me. Anyway, all good, Yi pointed it out in the code review too :)

**Pull request closed without merging**